sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -117,6 +119,33 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
         await this.educationProgramOfferingService.hasExistingOffering(
           programId,
         );
+
+      // In case it is an update, it is needed to know if the SABC code is being updated.
+      const persistedProgram = await this.getEducationProgramDetails(
```

Comment:
As a suggestion, if the method `hasExistingProgramWithSameSABCCode` received the `programId` it can execute the query for check duplicated in all records but the provided `programId`.

Reply:
ok. Done. Thanks!